contrib/babelfishpg_tsql/src/forjson.c

```diff
@@ -55,6 +57,31 @@ SPI_sql_row_to_json_path(uint64 rownum, StringInfo result, bool include_null_val
 	const char  *sep="";
 	bool 		isnull;
 
+	/*
+	* TODO: encode binary/varbinary/image data values using base64 encoding.
+	* Also, pg_b64_encode function might be useful.
+	* For now report an ERROR if any attribute is binary data type since base64
+	* encoding is not implemented yet.
+	* Also Exact numeric's datatype money or smallmoney is not supported.
+	*/
+	for (i = 1; i <= SPI_tuptable->tupdesc->natts; i++)
+	{
+		char* typename = SPI_gettype(SPI_tuptable->tupdesc, i);
+		if (strcmp(typename, "binary") == 0 ||
+			strcmp(typename, "varbinary") == 0 ||
+			strcmp(typename, "image") == 0 ||
+			strcmp(typename, "timestamp") == 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("option binary base64 is not supported")));
+		
+		if (strcmp(typename, "money") == 0 ||
+			strcmp(typename, "smallmoney") == 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("data type exact numeric's money or smallmoney is not supported")));
```

Comment:
message wording nitpick: leave it as "data types money or smallmoney are not supported with FOR JSON"